utilities/index.js

```diff
@@ -1,272 +1,315 @@
-
-      var redcap_json_questionnaires = splitIntoQuestionnaires(redcap_json);
-      var armt_json_questionnaires = splitIntoQuestionnaires(armt_json);
-
-      // Filter specified forms to update on Github
-      form_names = Object.keys(armt_json_questionnaires);
-      form_names = formNames.length
-        ? form_names.filter(v => formNames.includes(v))
-        : form_names;
+function postRADARJSON(
+  redcap_url,
+  redcap_token,
+  type,
+  langConvention,
+  formNames,
+) {
+  console.log('starting..');
+  var lang = langConvention || '';
+  var redcap_url = redcap_url || '';
+  var redcap_token = redcap_token || '';
+  var redcap_form_name = redcap_form_name || '';
+  var post_form = {
+    token: redcap_token,
+    content: 'metadata',
+    format: 'json',
+    returnFormat: 'json',
+    forms: [redcap_form_name],
+  };
 
-      console.log("Updating: " + form_names);  
-      
-      setInterval(function() {
-        if (globalItter == form_names.length) {
-          // Wait further 1500ms for asynchronous tasks to be completed
-          new Promise(resolve => setTimeout(resolve, 1500)).then(() => { process.exit()});
-          return;
-        }
+  request.post({ url: redcap_url, form: post_form }, function(
+    err,
+    httpResponse,
+    body,
+  ) {
+    var redcap_json = JSON.parse(body.replace(/(\r?\n|\r)/gm, '\n'));
+    var armt_json = REDCapConvertor(redcap_json);
+
+    var redcap_json_questionnaires = splitIntoQuestionnaires(redcap_json);
+    var armt_json_questionnaires = splitIntoQuestionnaires(armt_json);
+
+    // Filter specified forms to update on Github
+    form_names = Object.keys(armt_json_questionnaires);
+    form_names = formNames.length
+      ? form_names.filter(v => formNames.includes(v))
+      : form_names;
+
+    console.log('Updating: ' + form_names);
+
+    setInterval(function() {
+      if (globalItter == form_names.length) {
+        // Wait further 1500ms for asynchronous tasks to be completed
+        new Promise(resolve => setTimeout(resolve, 1500)).then(() => {
+          process.exit();
+        });
+        return;
+      }
 
-        form_name = form_names[globalItter];
-        form_armt = armt_json_questionnaires[form_name];
-        form_redcap = redcap_json_questionnaires[form_name];
+      form_name = form_names[globalItter];
+      form_armt = armt_json_questionnaires[form_name];
+      form_redcap = redcap_json_questionnaires[form_name];
 
-        // Remove if record_id field autogenerated by redcap exists
-        if(form_armt[0].field_name == 'record_id') 
-          form_armt.splice(0,1);
+      // Remove if record_id field autogenerated by redcap exists
+      if (form_armt[0].field_name == 'record_id') form_armt.splice(0, 1);
 
-        switch(type) {
-          case 'redcap': preparePostToGithub(form_name, form_redcap, lang);
+      switch (type) {
+        case 'redcap':
+          preparePostToGithub(form_name, form_redcap, lang);
```

Comment:
do we need a break here? or is it an intentional follow-through?

Reply:
I'm not sure, I think it was intended to post both Redcap DDs and questionnaire files to Github (so that after the case the default case will always be executed)? By the way for the data dictionary publishing, it hasn't been updated in 3 years, should that be updated or is that no longer needed in the script?